sw/device/lib/testing/kmac/BUILD

```diff
@@ -0,0 +1,11 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+cc_binary(
+    name = "kmac_vector_generate",
+    srcs = ["kmac_vector_generate.c"],
+    deps = [
+        "@xkcp//:nist_sp_800_185",
+    ],
+)
```

Comment:
I would rearchitect this into a library and add a `cc_test` target. That way, CI will warn you about compilation failures and unexpected results, if you encode expectations into the test!

If you intend to run it to extract results, you can either run the test directly or create a `cc_binary` that depends on the library and provides a `main()` function.

Reply:
I think what you said makes sense. I am not sure how to do this yet. Is converting `cc_binary` to `cc_test`, and adding some assertions to `kmac_vector_generate.c` does the trick? 

Or did you mean there should be some intermediate library `xyz`, which uses `@xkcp` as dependency as `xyz` is a dependency of `kmac_vector_generate`?

Eventually, it would be great if the test are automatically generated through XKCP and made available to KMAC library so that we can test this somewhere on CI. At least that's what I am hoping for.

Reply:
Since I removed the host binary from this PR, I added it to [TODO list](16410). I will come back to this later. Thanks @dmcardle.